components/Calender.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useEffect, useState } from 'react';
+import { View, Text, TouchableOpacity, Image } from 'react-native';
+import { CalenderStyles } from '@styles/CalenderStyles';
+import { LeftArrowIcon, RightArrowIcon } from '../icons';
+import { CalenderDays } from '@constants/CalenderConstants';
+import dayjs from 'dayjs';
+import { DateData, useLocal } from '../hooks/useLocal';
+
+interface Props {
+  streak: any;
+  pathId: number;
+  onStreakUpdate?: (streakValue: number) => void;
+}
+
+export const Calender = ({ pathId, streak, onStreakUpdate }: Props) => {
+  const [currentMonth, setCurrentMonth] = useState(dayjs());
+  const [days, setDays] = useState<string[]>([]);
+  const [progressDates, setProgressDates] = useState<DateData>();
+  const { fetchFromLocal } = useLocal();
+
+  const calculateStreak = (dates: string[]): number => {
```

Comment:
The calculateStreak function is called on every render and performs complex date calculations. Consider memoizing this with useMemo